app/routes/goals_routes.py

```diff
@@ -0,0 +1,118 @@
+from app import db
+from .tasks_routes import validate_model
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+
+goals_bp = Blueprint("goals_bp",__name__, url_prefix="/goals")
+
+@goals_bp.route("",methods=["POST"])
+def create_goal_2():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except:
+        return {"details": "Invalid data"} , 400
+    
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal":{
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }}, 201
+
+
+@goals_bp.route("",methods=["GET"])
+def read_all_goals():
+
+    goals = Goal.query.all()
+
+    goals_response = []
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_task(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return {
+        "goal":{
+        "id": goal.goal_id,
+        "title": goal.title}}, 200
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal_to_update = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal_to_update.title = request_body["title"]
+
+    db.session.commit()
+
+    return jsonify({"goal":goal_to_update.to_dict()}), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return abort(make_response({"details":f"Goal {goal_id} \"{goal.title}\" successfully deleted"}, 200))
+
+
+#RELATIONSHIP
+#Goals(parent) with tasks(child)
+#Como usuario, quiero agregar tasks a un goal (el goal te lo da el usuario en 
+# en la URL y como request_body te da la lista de tasks)
+
+#Preguntar porque POST si solo esta update????????????
```

Comment:
Right, it feels strange that we're making a `POST` request instead of `PUT` here because our goal and task records already exist and we're just updating our existing Goal (it says so on the method name even on L79!).

While `POST` is usually meant to create and `PUT` is usually meant to update, **both** `POST` and `PUT` can create and update resources. The true big difference between those two HTTP methods is a matter of [idempotency](https://www.restapitutorial.com/lessons/idempotency.html), an important computer concept we unfortunately don't really get into. There's still [a lot of debate among engineers](https://stackoverflow.com/questions/630453/what-is-the-difference-between-post-and-put-in-http) what HTTP method to use -- feel free to ask me more during 1:1!